src/plugins/tts/PageChunk.js

```diff
@@ -21,27 +21,27 @@ export default class PageChunk {
    * @param {number} leafIndex
    * @return {Promise<PageChunk[]>}
    */
-  static fetch(server, bookPath, leafIndex) {
+  static async fetch(server, bookPath, leafIndex) {
     // jquery's ajax "PromiseLike" implementation is inconsistent with
     // modern Promises, so convert it to a full promise (it doesn't forward
     // a returned promise to the next handler in the chain, which kind of
     // defeats the entire point of using promises to avoid "callback hell")
-    return new Promise((res, rej) => {
-      $.ajax({
+    try {
```

Comment:
Nice work! This was a tricky one.

Let's remove the try/catch. We want the error to go forward! Previously we forwarded it by calling the `reject` function. When constructing a promise like this, the `res` (short for `resolve`) function controls the returned value, and the `rej` (short for `reject`) function controls any errors thrown. In this case, we can just let `$.ajax` throw its error like normal :)